src/DataFrame/Operations/Statistics.hs

```diff
@@ -265,3 +263,34 @@ correlation' xs ys
              in go (i + 1) sumX' sumY' sumSquaredX' sumSquaredY' sumXY'
         | otherwise = (sumX, sumY, sumSquaredX, sumSquaredY, sumXY)
 {-# INLINE correlation' #-}
+
+quantiles' :: VU.Vector Int -> Int -> VU.Vector Double -> VU.Vector Double
+quantiles' qs q samp
+    | VU.null samp = throw $ EmptyDataSetException "quantiles"
+    | q < 2 = throw $ WrongQuantileNumberException q
+    | VU.any (\i -> i < 0 || i > q) qs = throw $ WrongQuantileIndexException qs q
+    | otherwise = runST $ do
+        let !n = VU.length samp
+        mutableSamp <- VU.thaw samp
+        VA.sort mutableSamp
+        sortedSamp <- VU.freeze mutableSamp
```

Comment:
When we freeze we pay an extra O(n) space and time cost. Since all we want to do here is index into a small subset of the vector freezing is wasteful. 

To make it more concrete, with an O(nlogn) sort on 100M elements of sorting takes 3s freezing will take another 0.5/1 second. 

So we should either:
* index the vector as is with the read function from the mutable package 
* use unsafeFreeze since we won't be returning the vector anyway.

Reply:
Actually...we can just fix this is a followup. I won't let it block this change.